src/erc20_utils.rs

```diff
@@ -36,11 +35,12 @@ impl Web3 {
                     "erc20 allowance(address, address) failed".to_string(),
                 ))
             }
-        });
+        })
+        .unwrap();
 
         // Check if the allowance remaining is greater than half of a Uint256- it's as good
         // a test as any.
-        Ok(allowance > (Uint256::max_value() / 2u32.into()))
+        Ok(allowance > Uint256::max_value().shr1())
```

Comment:
Does the logic here remain the same?

Reply:
`/ 2` may or may not be slow depending on how smart the compiler is. Because of the number of division and multiplication by two cases, I introduced `shr1` (shift right 1) and `shl1` (shift left 1) functions that do the same thing more performantly.